### PR TITLE
Update to upcoming 4.4 APIs

### PR DESCRIPTION
**Pull request by elmarco**

### CODE REVIEW

File: gdk4-wayland/Cargo.toml [hunk 1]
```diff
@@ -24,6 +25,8 @@ gio = {git = "https://github.com/gtk-rs/gtk-rs-core", features = ["v2_66"]}
 glib = {git = "https://github.com/gtk-rs/gtk-rs-core", features = ["v2_66"]}
 libc = "0.2"
 wayland-client = {version = "0.28", features = ["use_system_lib"]}
+khronos-egl = "4.1.0"
+xkb = "0.2.1"
```
Comment (bilelmoussaoui) [inline gdk4-wayland/Cargo.toml:29]: Those should be enabled only if v4_4 is selected?

File: gdk4-x11/Cargo.toml [hunk 2]
```diff
@@ -24,6 +25,7 @@ gio = {git = "https://github.com/gtk-rs/gtk-rs-core", features = ["v2_66"]}
 glib = {git = "https://github.com/gtk-rs/gtk-rs-core", features = ["v2_66"]}
 libc = "0.2"
 x11 = "2.18"
+khronos-egl = "4.1.0"
```
Comment (bilelmoussaoui) [inline gdk4-x11/Cargo.toml:28]: Same as for gdk-wayland, should only be used if the v4_4 feature is enabled

File: gsk4/src/auto/mod.rs [hunk 3]
```diff
@@ -2,13 +2,6 @@
 // from gir-files (https://github.com/gtk-rs/gir-files.git)
 // DO NOT EDIT
 
-#[cfg(any(feature = "broadway", feature = "dox"))]
```
Comment (bilelmoussaoui) [inline gsk4/src/auto/mod.rs:5]: Unexpected, you took the from a gtk build without the broadway backend enabled

File: gtk4/sys/src/lib.rs [hunk 4]
```diff
@@ -802,6 +802,7 @@ pub const GTK_ACCESSIBLE_VALUE_UNDEFINED: c_int = -1;
 pub const GTK_IM_MODULE_EXTENSION_POINT_NAME: *const c_char =
     b"gtk-im-module\0" as *const u8 as *const c_char;
 pub const GTK_INPUT_ERROR: c_int = -1;
+pub const GTK_INVALID_LIST_POSITION: c_int = 4294967295;
```
Comment (bilelmoussaoui) [inline gtk4/sys/src/lib.rs:805]: We should modify the one we use in gtk4/src/lib.rs to use the ffi value

File: gtk4/sys/src/lib.rs [hunk 5]
```diff
@@ -11301,6 +11330,7 @@ extern "C" {
     pub fn gtk_file_filter_add_mime_type(filter: *mut GtkFileFilter, mime_type: *const c_char);
     pub fn gtk_file_filter_add_pattern(filter: *mut GtkFileFilter, pattern: *const c_char);
     pub fn gtk_file_filter_add_pixbuf_formats(filter: *mut GtkFileFilter);
+    pub fn gtk_file_filter_add_suffix(filter: *mut GtkFileFilter, suffix: *const c_char);
```
Comment (bilelmoussaoui) [inline gtk4/sys/src/lib.rs:11333]: This seems to lack the since v4_4 annotation

File: gtk4/sys/src/lib.rs [hunk 6]
```diff
@@ -11963,6 +11997,7 @@ extern "C" {
     pub fn gtk_icon_theme_get_resource_path(self_: *mut GtkIconTheme) -> *mut *mut c_char;
     pub fn gtk_icon_theme_get_search_path(self_: *mut GtkIconTheme) -> *mut *mut c_char;
     pub fn gtk_icon_theme_get_theme_name(self_: *mut GtkIconTheme) -> *mut c_char;
+    pub fn gtk_icon_theme_has_gicon(self_: *mut GtkIconTheme, gicon: *mut gio::GIcon) -> gboolean;
```
Comment (bilelmoussaoui) [inline gtk4/sys/src/lib.rs:12000]: Same here

File: gtk4/sys/src/lib.rs [hunk 7]
```diff
@@ -12693,15 +12743,25 @@ extern "C" {
...
+    #[cfg(any(feature = "v4_4", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v4_4")))]
+    pub fn gtk_menu_button_get_always_show_arrow(menu_button: *mut GtkMenuButton) -> gboolean;
     pub fn gtk_menu_button_get_direction(menu_button: *mut GtkMenuButton) -> GtkArrowType;
     pub fn gtk_menu_button_get_has_frame(menu_button: *mut GtkMenuButton) -> gboolean;
     pub fn gtk_menu_button_get_icon_name(menu_button: *mut GtkMenuButton) -> *const c_char;
     pub fn gtk_menu_button_get_label(menu_button: *mut GtkMenuButton) -> *const c_char;
     pub fn gtk_menu_button_get_menu_model(menu_button: *mut GtkMenuButton) -> *mut gio::GMenuModel;
     pub fn gtk_menu_button_get_popover(menu_button: *mut GtkMenuButton) -> *mut GtkPopover;
+    pub fn gtk_menu_button_get_primary(menu_button: *mut GtkMenuButton) -> gboolean;
```
Comment (bilelmoussaoui) [inline gtk4/sys/src/lib.rs:12755]: and this

File: gtk4/sys/src/lib.rs [hunk 8]
```diff
@@ -13228,6 +13289,12 @@ extern "C" {
         show_peek_icon: gboolean,
     );
 
+    //=========================================================================
+    // GtkPasswordEntryBuffer
```
Comment (bilelmoussaoui) [inline gtk4/sys/src/lib.rs:13293]: Need to be added to the to be generated list in gtk4/Gir.toml

File: gdk4-wayland/Cargo.toml (hunk 1 above)
Comment (elmarco) [reply gdk4-wayland/Cargo.toml:29]: ack

File: gsk4/src/auto/mod.rs (hunk 3 above)
Comment (elmarco) [reply gsk4/src/auto/mod.rs:5]: Thanks I didn't noticed. I took it from a personal build. It seems when you last updated gir-files, you didn't update dl.sh script either, I assume you took it from a personal build too, right?

File: gtk4/sys/src/lib.rs (hunk 4 above)
Comment (elmarco) [reply gtk4/sys/src/lib.rs:805]: ok, missed that

File: gtk4/sys/src/lib.rs (hunk 5 above)
Comment (elmarco) [reply gtk4/sys/src/lib.rs:11333]: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/3736

File: gtk4/sys/src/lib.rs (hunk 8 above)
Comment (elmarco) [reply gtk4/sys/src/lib.rs:13293]: ack

File: gsk4/src/auto/mod.rs (hunk 3 above)
Comment (bilelmoussaoui) [reply gsk4/src/auto/mod.rs:5]: No, I use the gnome-nightly flatpak SDK, as it's assured to be built against latest releases/all the expected features enabled. You can enter the container with ` flatpak run --filesystem=home -d --command=sh org.gnome.Sdk//master` and then you can copy the gir files from their usual location at `/usr/share/gir-1.0/` to your home directory.

The dl.sh expects a distro having the latest release, we could maybe use fedora's gtk4-devel packages there but I don't plan on using it for gtk4

File: gtk4/sys/src/lib.rs (hunk 4 above)
Comment (elmarco) [reply gtk4/sys/src/lib.rs:805]: Had to fix GI: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/3738

File: gdk4-wayland/Cargo.toml [hunk 9]
```diff
@@ -24,6 +25,8 @@ gio = {git = "https://github.com/gtk-rs/gtk-rs-core", features = ["v2_66"]}
 glib = {git = "https://github.com/gtk-rs/gtk-rs-core", features = ["v2_66"]}
 libc = "0.2"
 wayland-client = {version = "0.28", features = ["use_system_lib"]}
+khronos-egl = {version = "4.1.0", optional = true}
+xkb = {version = "0.2.1", optional = true}
```
Comment (bilelmoussaoui) [inline gdk4-wayland/Cargo.toml:29]: The new dependencies's docs links needed to be added at https://github.com/gtk-rs/gtk4-rs/blob/master/.github/workflows/docs.yml#L38 for the docs to be properly linking to those crates

File: gtk4/Gir.toml [hunk 10]
```diff
@@ -1657,6 +1657,11 @@ name = "Gtk.PasswordEntry"
 status = "generate"
 trust_return_value_nullability = false
 
+[[object]]
+name = "Gtk.PasswordEntryBuffer"
+status = "generate"
+version = "4.4"
```
Comment (bilelmoussaoui) [inline gtk4/Gir.toml:1663]: I suppose this was upstreamed/reported?

File: gtk4/src/auto/menu_button.rs [hunk 11]
```diff
@@ -195,6 +234,48 @@ impl MenuButton {
...
+            glib::gobject_ffi::g_object_get_property(
+                self.as_ptr() as *mut glib::gobject_ffi::GObject,
+                b"primary\0".as_ptr() as *const _,
+                value.to_glib_none_mut().0,
+            );
+            value
+                .get()
+                .expect("Return Value for property `primary` getter")
+        }
+    }
```
Comment (bilelmoussaoui) [inline gtk4/src/auto/menu_button.rs:249]: There's a getter exposed already, this one should be disabled

File: gtk4/src/auto/menu_button.rs [hunk 12]
```diff
@@ -369,12 +473,16 @@ impl Default for MenuButton {
...
+    #[cfg(any(feature = "v4_4", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v4_4")))]
+    always_show_arrow: Option<bool>,
     direction: Option<ArrowType>,
     has_frame: Option<bool>,
     icon_name: Option<String>,
     label: Option<String>,
     menu_model: Option<gio::MenuModel>,
     popover: Option<Popover>,
+    primary: Option<bool>,
```
Comment (bilelmoussaoui) [inline gtk4/src/auto/menu_button.rs:485]: This lacks a since tag compared to `always_show_arrow`

File: gtk4/src/auto/stack_switcher.rs [hunk 13]
```diff
@@ -127,6 +129,7 @@ pub struct StackSwitcherBuilder {
     visible: Option<bool>,
     width_request: Option<i32>,
     accessible_role: Option<AccessibleRole>,
+    orientation: Option<Orientation>,
```
Comment (bilelmoussaoui) [inline gtk4/src/auto/stack_switcher.rs:132]: This one needs a since annotation as well

File: gtk4/sys/src/lib.rs [hunk 14]
```diff
@@ -12717,6 +12783,7 @@ extern "C" {
         menu_model: *mut gio::GMenuModel,
     );
     pub fn gtk_menu_button_set_popover(menu_button: *mut GtkMenuButton, popover: *mut GtkWidget);
+    pub fn gtk_menu_button_set_primary(menu_button: *mut GtkMenuButton, primary: gboolean);
```
Comment (bilelmoussaoui) [inline gtk4/sys/src/lib.rs:12786]: Another missing since annotation that went unnoticed in the latest review, sorry!

**Comment by bilelmoussaoui** [comment]
Looks good to me! Note that an update the gir-files repository is needed along with the CI that needs to use v4_4 features so that the new APIs is tested.  That might not work because the CI image is using 4.2.1 for now, a separate PR to bump that would be appreciated. See https://github.com/gtk-rs/gtk4-rs/blob/master/Dockerfile 
 
 Although, I will wait till I can get a 0.2 release out (by next week probably) first before merging v4_4 API in.

**Comment by bilelmoussaoui** [comment]
I forgot about https://github.com/gtk-rs/gtk4-rs/issues/494 but this can be removed from the manual code and generated automatically I think.

File: gtk4/src/auto/stack_switcher.rs (hunk 13 above)
Comment (elmarco) [reply gtk4/src/auto/stack_switcher.rs:132]: Hi @bilelmoussaoui , that one might be more tricky. https://gitlab.gnome.org/GNOME/gtk/-/commit/c848a5139537da8f6b90109598e30d6abcf30696 How do you annotate that a class got a new interface in a specific version?

Comment (sdroege) [reply gtk4/src/auto/stack_switcher.rs:132]: You can't currently, and also the `glib::wrapper!` macro can't express that currently

File: gtk4/Gir.toml (hunk 10 above)
Comment (elmarco) [reply gtk4/Gir.toml:1663]: ah yeah, it was not introspected correctly: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/3747

File: gdk4-wayland/Cargo.toml (hunk 9 above)
Comment (elmarco) [reply gdk4-wayland/Cargo.toml:29]: done

**Comment by elmarco** [comment]
> I forgot about #494 I suppose it can be removed from the manual code and generated automatically I think.

Looks like the existing binding is incorrect, as it doesn't pass a NULL-terminated array of char*.

File: gtk4/src/auto/menu_button.rs (hunk 12 above)
Comment (elmarco) [reply gtk4/src/auto/menu_button.rs:514]: fixed

File: gtk4/sys/src/lib.rs (hunk 14 above)
Comment (elmarco) [reply gtk4/sys/src/lib.rs:12788]: fixed

File: gdk4-wayland/Cargo.toml (hunk 9 above)
Comment (elmarco) [reply gdk4-wayland/Cargo.toml:29]: (I don't know how to verify it yet)

Comment (bilelmoussaoui) [reply gdk4-wayland/Cargo.toml:29]: Let's see how that goes once merged :)

File: gtk4/Gir.toml [hunk 15]
```diff
@@ -1661,6 +1657,10 @@ name = "Gtk.PasswordEntry"
 status = "generate"
 trust_return_value_nullability = false
 
+[[object]]
+name = "Gtk.PasswordEntryBuffer"
```
Comment (bilelmoussaoui) [inline gtk4/Gir.toml:1661]: No longer needed here now I guess

File: gtk4/src/auto/icon_theme.rs [hunk 16]
```diff
@@ -154,6 +166,13 @@ impl IconTheme {
         }
     }
 
+    #[doc(alias = "gtk_icon_theme_set_resource_path")]
+    pub fn set_resource_path(&self, path: &[&str]) {
```
Comment (bilelmoussaoui) [inline gtk4/src/auto/icon_theme.rs:170]: That's weird. It supposed to use `&[P]` with `P: AsRef<Path>`

File: gdk4-wayland/src/lib.rs [hunk 17]
```diff
@@ -14,6 +14,9 @@ pub use gio;
 pub use glib;
 pub use wayland_client;
 
+#[cfg(any(feature = "v4_4", feature = "dox"))]
+pub use xkb;
```
Comment (bilelmoussaoui) [inline gdk4-wayland/src/lib.rs:18]: but not khronos-egl?

File: gtk4/src/auto/stack_switcher.rs (hunk 13 above)
Comment (bilelmoussaoui) [reply gtk4/src/auto/stack_switcher.rs:132]: I wouldn't bother for now. Just keep a comment on GTK's Gir.toml file on StackSwitcher's entry

File: Dockerfile [hunk 18]
```diff
@@ -5,9 +5,9 @@ RUN dnf install git xorg-x11-server-Xvfb procps-ng wget 'dnf-command(builddep)'
 RUN dnf builddep gtk4 -y
 
 # build gtk4 from the latest release
-ADD https://download.gnome.org/sources/gtk/4.2/gtk-4.2.1.tar.xz /tmp/gtk-4.2.1.tar.xz
-RUN tar -xf /tmp/gtk-4.2.1.tar.xz --directory /tmp
-WORKDIR /tmp/gtk-4.2.1
+ADD https://download.gnome.org/sources/gtk/4.3/gtk-4.3.2.tar.xz /tmp/gtk-4.3.2.tar.xz
+RUN tar -xf /tmp/gtk-4.3.2.tar.xz --directory /tmp
+WORKDIR /tmp/gtk-4.3.2
```
Comment (bilelmoussaoui) [inline Dockerfile:10]: Please do this in a separate PR so we can merge it already. Otherwise the CI will fail because of the outdated image

File: gsk4/src/auto/mod.rs (hunk 3 above)
Comment (elmarco) [reply gsk4/src/auto/mod.rs]: done in https://github.com/gtk-rs/gir-files/pull/82

File: gtk4/Gir.toml (hunk 15 above)
Comment (elmarco) [reply gtk4/Gir.toml:1663]: fixed

File: gdk4-wayland/src/lib.rs (hunk 17 above)
Comment (elmarco) [reply gdk4-wayland/src/lib.rs:18]: done

File: Dockerfile (hunk 18 above)
Comment (bilelmoussaoui) [reply Dockerfile:10]: The image should be ready once this job https://github.com/gtk-rs/gtk4-rs/runs/3172466469 is over

File: gtk4/src/auto/icon_theme.rs (hunk 16 above)
Comment (elmarco) [reply gtk4/src/auto/icon_theme.rs:170]: no, the default is "/org/gtk/libgtk/icons/", so that binding looks correct to me.

File: gtk4/src/auto/stack_switcher.rs (hunk 13 above)
Comment (elmarco) [reply gtk4/src/auto/stack_switcher.rs:132]: done

File: gtk4/src/auto/icon_theme.rs (hunk 16 above)
Comment (bilelmoussaoui) [reply gtk4/src/auto/icon_theme.rs:170]: Never mind me, I got confused by the `_path` part of the function name.

**Comment by elmarco** [comment]
At least two issues for me left:
- how to lower the pkg-config version requirement to 4.3.2 ?
- in `gdk4-wayland/sys/Cargo.toml`, how to keep `v4_4 = ["xkbcommon-sys"]` from being overwritten on regen?

File: gdk4-wayland/sys/Cargo.toml [hunk 19]
```diff
@@ -13,6 +13,9 @@ version = "0.2.0"
 [package.metadata.system-deps.gtk4_wayland]
 name = "gtk4-wayland"
 version = "4.0.0"
+
+[package.metadata.system-deps.gtk4_wayland.v4_4]
+version = "4.4"
```
Comment (bilelmoussaoui) [inline gdk4-wayland/sys/Cargo.toml:18]: here's where you can change the required version number

**Comment by bilelmoussaoui** [comment]
For me there are still two things as well:
- Bump the feature used in the CI to the latest ones https://github.com/gtk-rs/gtk4-rs/blob/master/.github/workflows/CI.yml#L22 & mention the new `v4_4` feature in the various readmes. 
- The vulkan feature is still used in the CI, in the source code of gsk as well. All of that needs to be dropped

**Comment by bilelmoussaoui** [comment]
Ideally the new features in gdk-x11/gdk-wayland would have to be documented in the readme as well

**Comment by bilelmoussaoui** [comment]
>     * in `gdk4-wayland/sys/Cargo.toml`, how to keep `v4_4 = ["xkbcommon-sys"]` from being overwritten on regen?

This might help https://github.com/gtk-rs/gtk3-rs/blob/master/gdkx11/sys/Gir.toml#L18-L22 maybe?

File: gdk4-wayland/sys/Cargo.toml (hunk 19 above)
Comment (elmarco) [reply gdk4-wayland/sys/Cargo.toml:18]: That's what I thought, but thought it was generated. it's magic it isn't, or not fully :)

**Comment by elmarco** [comment]
> This might help https://github.com/gtk-rs/gtk3-rs/blob/master/gdkx11/sys/Gir.toml#L18-L22 maybe?
it works! thanks

**Comment by elmarco** [comment]
Alright, let see if I missed anything else.

File: gdk4-wayland/sys/src/manual.rs [hunk 20]
```diff
@@ -0,0 +1,4 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+#[cfg(any(feature = "v4_4", feature = "dox"))]
```
Comment (bilelmoussaoui) [inline gdk4-wayland/sys/src/manual.rs:3]: I don't think you need the feature="dox" here, it's what causes the docs job to fail

File: gdk4/sys/src/lib.rs [hunk 21]
```diff
@@ -4209,6 +4218,8 @@ extern "C" {
         minor: *mut c_int,
     );
     pub fn gdk_gl_context_is_legacy(context: *mut GdkGLContext) -> gboolean;
+    pub fn gdk_gl_context_is_shared(self_: *mut GdkGLContext, other: *mut GdkGLContext)
```
Comment (bilelmoussaoui) [inline gdk4/sys/src/lib.rs:4221]: Needs a since annotation

Comment (elmarco) [reply gdk4/sys/src/lib.rs:4221]: crap, that went unnoticed in 4.3.2.. https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/3790
How do I override that in gir now? :)

Comment (bilelmoussaoui) [reply gdk4/sys/src/lib.rs:4221]: You can override the version needed for the function & add a comment to drop it next gir files bump

Comment (elmarco) [reply gdk4/sys/src/lib.rs:4221]: done

File: gdk4-wayland/sys/src/manual.rs (hunk 20 above)
Comment (elmarco) [reply gdk4-wayland/sys/src/manual.rs:3]: I think we were just missing some doc_cfg feature on the crate. Should be fixed.

File: gtk4/Cargo.toml [hunk 22]
```diff
@@ -18,6 +18,7 @@ name = "gtk4"
 [features]
 dox = ["gdk/dox", "gsk/dox", "ffi/dox"]
 v4_2 = ["ffi/v4_2", "gdk/v4_2", "gsk/v4_2"]
+v4_4 = ["ffi/v4_4", "gdk/v4_4"]
```
Comment (bilelmoussaoui) [inline gtk4/Cargo.toml:21]:
```suggestion
v4_4 = ["v4_2", "ffi/v4_4", "gdk/v4_4"]
```
v4_4 should enable v4_2 apis as well

File: gdk4-x11/Cargo.toml [hunk 23]
```diff
@@ -12,7 +12,8 @@ repository = "https://github.com/gtk-rs/gtk4-rs"
 version = "0.2.0"
 
 [features]
-dox = ["ffi/dox"]
+v4_4 = ["ffi/v4_4", "khronos-egl"]
+dox = ["ffi/dox", "v4_4"]
```
Comment (bilelmoussaoui) [inline gdk4-x11/Cargo.toml:16]:
```suggestion
dox = ["ffi/dox"]
```
this doesn't make much sense, dox is a workaround around having to enable all the futures.

File: gdk4-wayland/sys/Cargo.toml [hunk 24]
```diff
@@ -13,18 +13,26 @@ version = "0.2.0"
...
+version = "4.3.2"
 [package.metadata.docs.rs]
 features = ["dox"]
 
 [lib]
 name = "gdk4_wayland_sys"
 
 [features]
-dox = []
+dox = ["xkbcommon-sys"]
```
Comment (bilelmoussaoui) [inline gdk4-wayland/sys/Cargo.toml:26]:
```suggestion
dox = []
```
is this needed?

File: .github/workflows/CI.yml [hunk 25]
```diff
@@ -19,11 +19,11 @@ jobs:
           - nightly
           - "1.53.0"
         conf:
-          - { name: "gtk4", features: "v4_2", test_sys: false }
-          - { name: "gsk4", features: "v4_2,broadway,vulkan", test_sys: true }
-          - { name: "gdk4", features: "v4_2", test_sys: true }
-          - { name: "gdk4-wayland", features: "", test_sys: true }
-          - { name: "gdk4-x11", features: "", test_sys: false }
+          - { name: "gtk4", features: "v4_2,v4_4", test_sys: false }
```
Comment (bilelmoussaoui) [inline .github/workflows/CI.yml:22]:
```suggestion
          - { name: "gtk4", features: "v4_4", test_sys: false }
```
You can drop v4_2 from the ones with v4_4

**Comment by bilelmoussaoui** [comment]
Looks generally good to me. Let's see what @sdroege thinks

File: gdk4/src/auto/draw_context.rs [hunk 26]
```diff
@@ -85,28 +77,6 @@ impl<O: IsA<DrawContext>> DrawContextExt for O {
             ))
         }
     }
-
-    fn connect_display_notify<F: Fn(&Self) + 'static>(&self, f: F) -> SignalHandlerId {
```
Comment (sdroege) [inline gdk4/src/auto/draw_context.rs:89]: Why is this removed?

File: gdk4/src/auto/texture.rs [hunk 27]
```diff
@@ -76,11 +76,11 @@ impl<O: IsA<Texture>> TextureExt for O {
         unsafe { ffi::gdk_texture_get_width(self.as_ref().to_glib_none().0) }
     }
 
-    fn save_to_png(&self, filename: &str) -> bool {
+    fn save_to_png<P: AsRef<std::path::Path>>(&self, filename: P) -> bool {
```
Comment (sdroege) [inline gdk4/src/auto/texture.rs:79]: Needs to become 0.3 because of API changes @bilelmoussaoui (just so you don't forget :) )

File: gtk4/src/icon_theme.rs [hunk 28]
```diff
@@ -1,13 +0,0 @@
-// Take a look at the license at the top of the repository in the LICENSE file.
-
-use crate::IconTheme;
-use glib::translate::*;
-
-impl IconTheme {
-    #[doc(alias = "gtk_icon_theme_set_resource_path")]
-    pub fn set_resource_path(&self, path: &str) {
```
Comment (sdroege) [inline gtk4/src/icon_theme.rs:8]: Why did this disappear?

**Review by sdroege — COMMENTED** [review]
Looks generally good to me. Are there any other new objects that would be useful to bind? If so please create an issue for them :)

File: gtk4/Cargo.toml (hunk 22 above)
Comment (elmarco) [reply gtk4/Cargo.toml:21]: done

File: gdk4-x11/Cargo.toml (hunk 23 above)
Comment (elmarco) [reply gdk4-x11/Cargo.toml:16]: Without it, doc building fails:
```
error[E0432]: unresolved import `khronos_egl`
 --> gdk4-x11/src/x11_display.rs:6:5
  |
6 | use khronos_egl as egl;
  |     ^^^^^^^^^^^^^^^^^^ no external crate `khronos_egl`

error[E0432]: unresolved import `xkb`
 --> gdk4-wayland/src/wayland_device.rs:9:5
  |
9 | use xkb::Keymap;
  |     ^^^ use of undeclared crate or module `xkb`

error[E0432]: unresolved import `khronos_egl`
 --> gdk4-wayland/src/wayland_display.rs:6:5
  |
6 | use khronos_egl as egl;
  |     ^^^^^^^^^^^^^^^^^^ no external crate `khronos_egl`

error[E0432]: unresolved import `khronos_egl`
  --> gdk4-wayland/src/lib.rs:20:9
   |
20 | pub use khronos_egl;
   |         ^^^^^^^^^^^ no external crate `khronos_egl`

error[E0432]: unresolved import `xkb`
  --> gdk4-wayland/src/lib.rs:23:9
   |
23 | pub use xkb;
   |         ^^^ no external crate `xkb`
```

Should it exclude those related API when generating docs?

File: gdk4-wayland/sys/Cargo.toml (hunk 24 above)
Comment (elmarco) [reply gdk4-wayland/sys/Cargo.toml:26]: Needed to generate doc.

File: .github/workflows/CI.yml (hunk 25 above)
Comment (elmarco) [reply .github/workflows/CI.yml:22]: ah ok, done

File: gdk4/src/auto/draw_context.rs (hunk 26 above)
Comment (elmarco) [reply gdk4/src/auto/draw_context.rs:89]:
```
-      <property name="display" transfer-ownership="none">
+      <property name="display" writable="1" construct-only="1" transfer-ownership="none">
```

File: gtk4/src/icon_theme.rs (hunk 28 above)
Comment (elmarco) [reply gtk4/src/icon_theme.rs:8]: See `gtk: generate and fix set_resource_path()`. I am not even clear whether the current binding actually work, since the strv argument may not be NULL-terminated.

**Comment by bilelmoussaoui** [comment]
> Looks generally good to me. Are there any other new objects that would be useful to bind? If so please create an issue for them :)

The only new one is PasswordEntryBuffer and it was already added

File: gdk4-x11/Cargo.toml (hunk 23 above)
Comment (bilelmoussaoui) [reply gdk4-x11/Cargo.toml:16]: Yes, because you have the same issue on those as well. The `dox` feature should be dropped from imports

Comment (elmarco) [reply gdk4-x11/Cargo.toml:16]: But the API depend on those crate.. I am lost. Please show me :)

File: gdk4-wayland/src/lib.rs [hunk 29]
```diff
@@ -8,12 +8,20 @@
...
+#![cfg_attr(feature = "dox", feature(doc_cfg))]
+
 pub use ffi;
 pub use gdk;
 pub use gio;
 pub use glib;
 pub use wayland_client;
 
+#[cfg(any(feature = "v4_4", feature = "dox"))]
+pub use khronos_egl;
```
Comment (bilelmoussaoui) [inline gdk4-wayland/src/lib.rs:20]:
```suggestion
#[cfg(feature = "v4_4"))]
pub use khronos_egl;
```
I meant something like this

Comment (elmarco) [reply gdk4-wayland/src/lib.rs:20]: But it'll fail with the API that use it.

**Comment by bilelmoussaoui** [comment]
Sorry for taking too long, i will do a 0.2.1 & then we can merge this so we can target 0.3.

File: gdk4-wayland/Cargo.toml (hunk 9 above)
Comment (elmarco) [reply gdk4-wayland/Cargo.toml:29]: resolving, assuming it works for now

**Comment by elmarco** [comment]
@bilelmoussaoui @sdroege anything else I can do? thanks

**Comment by sdroege** [comment]
Ok from my side if it's OK for @bilelmoussaoui :)

**Review by sdroege — APPROVED** [review]

**Comment by bilelmoussaoui** [comment]
Overall looks good to me, but it needs to be rebased before it can be merged
